File: benchmarks/solutions/rust/a1-two-sum.rs
use std::collections::HashMap;

fn two_sum(nums: &[i32], target: i32) -> (usize, usize) {
    let mut seen = HashMap::new();
    for (i, &n) in nums.iter().enumerate() {
        let complement = target - n;
        if let Some(&j) = seen.get(&complement) {
            return (j, i);
        }
        seen.insert(n, i);
    }
    unreachable!()
}

File: benchmarks/solutions/objc/a1-two-sum.m
#import <Foundation/Foundation.h>

NSArray<NSNumber *> *twoSum(NSArray<NSNumber *> *nums, NSInteger target) {
    NSMutableDictionary<NSNumber *, NSNumber *> *seen = [NSMutableDictionary dictionary];
    for (NSUInteger i = 0; i < nums.count; i++) {
        NSInteger n = nums[i].integerValue;
        NSInteger complement = target - n;
        NSNumber *j = seen[@(complement)];
        if (j != nil) {
            return @[j, @(i)];
        }
        seen[@(n)] = @(i);
    }
    return @[@(-1), @(-1)];
}
